frontend/controller/actions/group.js

```
@@ -13,12 +18,31 @@ import {
 } from '@model/contracts/voting/constants.js'
 import { GIErrorUIRuntimeError } from '@model/errors.js'
 import { imageUpload } from '@utils/image.js'
-import { merge } from '@utils/giLodash.js'
+import { merge, omit } from '@utils/giLodash.js'
 import { dateToPeriodStamp, addTimeToDate, DAYS_MILLIS } from '@utils/time.js'
 import L, { LError } from '@view-utils/translations.js'
 import { encryptedAction } from './utils.js'
 import type { GIActionParams } from './types.js'
 
+export async function leaveAllChatRooms (groupContractID: string, member: string) {
+  // let user leaves all the chatrooms before leaving group
+  const rootState = sbp('state/vuex/state')
+  const chatRooms = rootState[groupContractID].chatRooms
+  const chatRoomIDsToLeave = Object.keys(chatRooms)
+    .filter(cID => chatRooms[cID].users.includes(member))
+
+  try {
+    for (const chatRoomID of chatRoomIDsToLeave) {
+      await sbp('gi.actions/group/leaveChatRoom', {
+        contractID: groupContractID,
+        data: { chatRoomID, member, leavingGroup: true }
+      })
+    }
+  } catch (e) {
+    throw new GIErrorUIRuntimeError(L('Failed to leave chat channel.'))
```

This hides the original error. At the very least, above this line should be:

```js
console.error(`leaveAllChatRooms: ${e.name}: ${e.message}`, e)
```